### PR TITLE
fix: Disabled input should not trigger the action of suffix part #18889

### LINKED ISSUE #18889: Input's Search component also triggers the onSearch event when disabled is true
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-h5ubd?fontsize=14)

### Steps to reproduce
1. set disabled in Input.Search component
2. click search icon

### What is expected?
1. Search icon hover style should not change when disabled
2. onSearch event will not be triggered

### What is actually happening?
1. Search icon hover style will change when disabled
2. onSearch event will be triggered

| Environment | Info |
|---|---|
| antd | 3.23.3 |
| React | react |
| System | win10 |
| Browser | chrome v76.0.3809.132 |

### PR DESCRIPTION
**Pull request by dilidili**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close #18889 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Disabled input should not trigger the action of suffix part     |
| 🇨🇳 Chinese |    失效状态的 Input  的后缀图标应不可点击     |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
来个用例如何？

**Comment by dilidili** [comment]
> 来个用例如何？

好了加上了

**Comment by yoyo837** [comment]
用例的`react/jsx-boolean-value` 处理下

File: components/input/__tests__/Password.test.js [hunk 1]
```diff
@@ -40,6 +40,13 @@ describe('Input.Password', () => {
     expect(wrapper.find('.anticon-eye-invisible').length).toBe(1);
   });
 
+  it('should not toggle visibility when disabled prop is true', () => {
+    const wrapper = mount(<Input.Password disabled={true} />);
```
Comment (yoyo837) [inline components/input/__tests__/Password.test.js]: `react/jsx-boolean-value`

File: components/input/__tests__/Search.test.js [hunk 2]
```diff
@@ -31,6 +31,15 @@ describe('Input.Search', () => {
     expect(wrapper.find('.ant-btn-primary[disabled]')).toHaveLength(1);
   });
 
+  it('should disable search icon when disabled prop is true', () => {
+    const onSearch = jest.fn();
+    const wrapper = mount(
+      <Search defaultValue="search text" onSearch={onSearch} disabled={true} />,
```
Comment (yoyo837) [inline components/input/__tests__/Search.test.js]: `react/jsx-boolean-value`

**Comment by JounQin** [comment]
Password 类型 disabled 状态是否应该可以切换可见性，还是应该用 `readOnly`？

**Pull request merged**